clients/admin-ui/src/features/test-datasets/DatasetEditorSection.tsx

```diff
@@ -105,28 +105,43 @@ const EditorSection = ({ connectionKey }: EditorSectionProps) => {
   };
 
   const handleSave = async () => {
+    if (!currentDataset) {
+      return;
+    }
+
+    // Parse YAML first
+    let datasetValues: Dataset;
     try {
-      if (currentDataset) {
-        const datasetValues = yaml.load(editorContent) as Partial<Dataset>;
-        const updatedDatasetConfig: DatasetConfigSchema = {
-          fides_key: currentDataset.fides_key,
-          ctl_dataset: {
-            ...currentDataset.ctl_dataset,
-            ...datasetValues,
-          },
-        };
```

Comment:
This was causing some weird consolidation issues, it was better to just pass the `datasetValues` directly to the `updateDataset`